carlsim/test/cuba.cpp

```diff
@@ -152,66 +153,208 @@ TEST(CUBA, firingRateCPUvsGPU) {
-			bool isRateCorrectG1 = spkMonG1->getPopMeanFiringRate() == spkRateG1CPU;
-			EXPECT_TRUE(isRateCorrectG0);
-			EXPECT_TRUE(isRateCorrectG1);
-
-			if (isRateCorrectG0 && isRateCorrectG1) {
-				spkTimesG0GPU = spkMonG0->getSpikeVector2D();
-				spkTimesG1GPU = spkMonG1->getSpikeVector2D();
-				bool isSpkSzCorrectG0 = spkTimesG0CPU[0].size() == spkTimesG0GPU[0].size();
-				bool isSpkSzCorrectG1 = spkTimesG1CPU[0].size() == spkTimesG1GPU[0].size();
-				EXPECT_TRUE(isSpkSzCorrectG0);
-				EXPECT_TRUE(isSpkSzCorrectG1);
-
-				if (isSpkSzCorrectG0 && isSpkSzCorrectG1) {
-					for (int i=0; i<spkTimesG0CPU[0].size(); i++)
-						EXPECT_EQ(spkTimesG0CPU[0][i], spkTimesG0GPU[0][i]);
-					for (int i=0; i<spkTimesG1CPU[0].size(); i++)
-						EXPECT_EQ(spkTimesG1CPU[0][i], spkTimesG1GPU[0][i]);
+					if (isRateCorrectG0 && isRateCorrectG1) {
+						spkTimesG0GPU = spkMonG0->getSpikeVector2D();
+						spkTimesG1GPU = spkMonG1->getSpikeVector2D();
+						bool isSpkSzCorrectG0 = spkTimesG0CPU[0].size() == spkTimesG0GPU[0].size();
+						bool isSpkSzCorrectG1 = spkTimesG1CPU[0].size() == spkTimesG1GPU[0].size();
+						EXPECT_TRUE(isSpkSzCorrectG0);
+						EXPECT_TRUE(isSpkSzCorrectG1);
+
+						if (isSpkSzCorrectG0 && isSpkSzCorrectG1) {
+							for (int i = 0; i<spkTimesG0CPU[0].size(); i++)
+								EXPECT_EQ(spkTimesG0CPU[0][i], spkTimesG0GPU[0][i]);
+							for (int i = 0; i<spkTimesG1CPU[0].size(); i++)
+								EXPECT_EQ(spkTimesG1CPU[0][i], spkTimesG1GPU[0][i]);
+						}
+					}
 				}
 			}
 		}
 	}
 }
+
 #endif // !__NO_CUDA__
+
+TEST(CUBA, firingRateVsData9Param) {
+	::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+#ifdef __NO_CUDA__
+	int numModes = 1;
+#else
+	int numModes = 2;
+#endif
+
+	for (int isGPUmode = 0; isGPUmode<numModes; isGPUmode++) {
+		CARLsim* sim = new CARLsim("CUBA.firingRateVsData9Param", isGPUmode ? GPU_MODE : CPU_MODE, SILENT, 0, 42);
+		int g1 = sim->createGroup("excit", 1, EXCITATORY_NEURON);
+		sim->setNeuronParameters(g1, 100.0f, 0.7f, -60.0f, -40.0f, 0.03f, -2.0f, 35.0f, -50.0f, 100.0f);
+		int g0 = sim->createSpikeGeneratorGroup("input", 1, EXCITATORY_NEURON);
+
+		sim->setConductances(false); // make CUBA explicit
+
+		sim->connect(g0, g1, "one-to-one", RangeWeight(0.0f), 1.0f, RangeDelay(1), RadiusRF(-1));
+
+		sim->setupNetwork();
+
+		PoissonRate in(1);
+		in.setRates(0.0f);
+		sim->setSpikeRate(g0, &in);
+
+		SpikeMonitor *spkMonG1 = sim->setSpikeMonitor(g1, "NULL");
+
+		spkMonG1->startRecording();
+		sim->setExternalCurrent(g1, 0);
+		sim->runNetwork(0, 100);
+		sim->setExternalCurrent(g1, 70);
+		sim->runNetwork(0, 900);
+		spkMonG1->stopRecording();
+
+		//Data is taken from page 275 of Dynamical Systems in Neuroscience The Geometry of Excitability and Bursting
+		//by Eugene M. Izhikevich
+		EXPECT_FLOAT_EQ(spkMonG1->getPopMeanFiringRate(), 6.0f);
+		std::vector<std::vector<int> > spikeTimes = spkMonG1->getSpikeVector2D();
+		EXPECT_GT(spikeTimes[0][0], 195);
```

Comment:
Spike times should be fixed given no randomness. Use EXPECT_EQ() instead. The timing maybe a little bit off on different platforms.